### ISSUE TITLE
Syntax typo in nuget Auditing packages documentation

### ISSUE THREAD

**Issue by asdm90** [NuGet/docs.microsoft.com-nuget#3285]
I think I have found a typo or error in the documentation in the use of the WarningsNotAsErrors tag when auditing nuget packages.

The documentation on https://learn.microsoft.com/en-us/nuget/concepts/auditing-packages#warning-codes states that the syntax for WarningsNotAsErrors is semicolon-separated:

<WarningsNotAsErrors>NU1903;NU1904</WarningsNotAsErrors>

But in the linked documentation for WarningsNotAsErrors tag https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/errors-warnings#warningsaserrors-and-warningsnotaserrors the syntax is shown to be comma-separated:

<WarningsNotAsErrors>NU1903,NU1904</WarningsNotAsErrors>

The last one is the correct syntax, so I think the auditing documentation should be updated to follow the other documentation. Otherwise people might get confused to see that NU1903 warnings are still displayed as errors and therefore fails the build when running with <TreatWarningsAsErrors>true</TreatWarningsAsErrors>.

**Comment by nkolev92** [comment]
I think the compiler uses `,`, so both are supported by the tooling. 

If you look at https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022, it does say NoWarn is semicolon separated. 

The NuGet tooling splits by both.

**Comment by asdm90** [comment]
> I think the compiler uses `,`, so both are supported by the tooling.
> 
> If you look at https://learn.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2022, it does say NoWarn is semicolon separated.
> 
> The NuGet tooling splits by both.

I was not talking about the NoWarn tag but the WarningsNotAsErrors tag. They might be using the same conventions but I found in the project I am working on, putting the following lines inside a solution level Directory.Build.props file does NOT work with semicolons as separator for entries in the tag WarningsNotAsErrors but does work when using commas:

`<PropertyGroup>
  <TargetFramework>net8.0</TargetFramework>
  <ImplicitUsings>enable</ImplicitUsings>
  <Nullable>enable</Nullable>
  <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
  <CheckEolTargetFramework>true</CheckEolTargetFramework>
  <NuGetAudit>true</NuGetAudit>
  <NuGetAuditMode>all</NuGetAuditMode>
  <NuGetAuditLevel>low</NuGetAuditLevel>
  <WarningsNotAsErrors>NU1900,NU1901,NU1902,NU1903,NU1904,NETSDK1138</WarningsNotAsErrors>
</PropertyGroup>`

Now, whether this is because of an error in the tooling or in the documentation i cannot tell, but it seems worth investigating further.

**Comment by zivkan** [comment]
NuGet calls a static method named `MSBuildStringUtility.GetNuGetLogCodes` to read `WarningsAsErrors`, `NoWarn`, and `WarningsNotAsErrors`: https://github.com/NuGet/NuGet.Client/blob/4020570fd78096035b551a5ac0869ca995f4dbed/src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs#L232-L234

`GetNuGetLogCodes` splits on both `;` and `,`: https://github.com/NuGet/NuGet.Client/blob/4020570fd78096035b551a5ac0869ca995f4dbed/src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs#L95

I tried a quick search of roslyn's source to try to figure out how they split NoWarn, WarningsAsErrors and WarningsNotAsErrors, but I wasn't able to quickly find the relevant code.

Semicolons have been a convention in MSBuild, for as long as I've been working at Microsoft on NuGet, for splitting multiple values, and in certain contexts (like defining items), only semicolons are allowed to delimit multiple values: https://learn.microsoft.com/en-us/visualstudio/msbuild/msbuild-items?view=vs-2022#create-items-in-a-project-file

I assume support for `,` separators was added since it's commonly used in other programming languages, but if Roslyn was already allowing commas before NuGet added support for NoWarn, then the NuGet team at the time didn't really have a choice but to support commas as well, since NoWarn and WarningsAsErrors were already a feature in Roslyn.

TL;DR: both commas as semicolons should work. If they're not, something is probably being done wrong, but a sample will help investigate if it's a product bug. However, I did explicitly test these properties when I implemented NuGetAudit, so I'm reasonably confident they should work

**Comment by asdm90** [comment]
Maybe it is in the interaction with Central Package Management. I will see if I can put together an example project with the error to investigate it.
